src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -163,14 +164,16 @@ public void onFailure(IOException e) {
             try {
                 Thread.sleep(1000);
             } catch (InterruptedException e) {
+                // need to throw as this indicates the step as been cancelled
+                throw e;
```

Comment:
we should also invoke a cancel build action to cancel the build running on openshift.

ditto for the deployment step.